ESP/I2C_ESP32/main/access_buffers.cpp

```diff
@@ -0,0 +1,232 @@
+
+//     int ret;
+//     uint32_t task_idx = (uint32_t)taskID;
+
+//     int dataLength = 64;
+//     uint8_t data[dataLength];
+
+//     getDefaultArray(data, dataLength);
+
+//     int count = 0;
+
+//     while (1) {
+//         std::cout << "==================\n";
+//         std::cout << "test count: " << count++ << "\n";
+//         std::cout << "==================\n";
+
+//         ret = master.write(data, master.getRwBufferLength());
+
+//         if (ret == ESP_ERR_TIMEOUT) {
+//             std::cout << "I2C TIME_OUT\n";
+//         } else if (ret == ESP_OK) {
+//             std::cout << "Task[" << task_idx << "] master write:\n";
+//             master.printBuffer(master.getTxBuffer(), master.getRwBufferLength());
+//         } else {
+//             std::cout << "Task [" << task_idx << "] " << esp_err_to_name(ret)
+//                       << ": Master write slave error, IO not connected...\n";
+//         }
+
+//         ret = master.read();
+
+//         if (ret == ESP_ERR_TIMEOUT) {
+//             std::cout << "I2C TIME_OUT\n";
+//         } else if (ret == ESP_OK) {
+//             std::cout << "TASK[" << task_idx << "] master read:\n";
+//             master.printBuffer(master.getRxBuffer(), master.getRwBufferLength());
+//         } else {
+//             std::cout << esp_err_to_name(ret) << ": Master read slave error, IO not connected...\n";
+//         }
+
+//         vTaskDelay((delayTimeBetweenItemsMS * (task_idx + 1)) / portTICK_RATE_MS);
+//     }
+// }
+
+uint8_t *AccessBuffers::slaveRead(void *taskID) {
+    i2c_port_t slavePortNum = i2c_port_t::I2C_NUM_0;   ///< Slave port number
+    gpio_num_t slaveSDA     = gpio_num_t::GPIO_NUM_25; ///< GPIO number for slave dataBuffer
+    gpio_num_t slaveSCL     = gpio_num_t::GPIO_NUM_26; ///< GPIO number for slave CLK
+
+    I2cEsp slave(slaveSDA, slaveSCL, slavePortNum);
+
+    uint32_t task_idx = (uint32_t)taskID;
+
+    int size = 0;
+
+    size = slave.read();
+
+    if (size == ESP_ERR_TIMEOUT) {
+        int bufferLength = 64; // just an random size
+        uint8_t buffer[bufferLength];
+        getDefaultArray(buffer, bufferLength);
+        return buffer;
+    } else {
+        return slave.getDataBuffer();
+    }
+    vTaskDelay((delayTimeBetweenItemsMS * (task_idx + 1)) / portTICK_RATE_MS);
+}
+
+void AccessBuffers::slaveWrite(uint8_t *buffer, int bufferLength, void *taskID) {
+    i2c_port_t slavePortNum = i2c_port_t::I2C_NUM_0;   ///< Slave port number
+    gpio_num_t slaveSDA     = gpio_num_t::GPIO_NUM_25; ///< GPIO number for slave dataBuffer
+    gpio_num_t slaveSCL     = gpio_num_t::GPIO_NUM_26; ///< GPIO number for slave CLK
+
+    I2cEsp slave(slaveSDA, slaveSCL, slavePortNum);
+
+    uint32_t task_idx = (uint32_t)taskID;
+
+    size_t d_size = 0;
+    int x         = slave.getRwBufferLength();
+
+    if (bufferLength > x) {
+        std::cout << "Size of parameter buffer to big\n";
+        return;
```

Comment:
Propper error should be given not only a cout. This makes it impossible for other functions to check if this function was successfull.